server/src/feedback/tokens.rs

```diff
@@ -0,0 +1,162 @@
+use actix_web::web::Data;
+use actix_web::HttpResponse;
+
+use jsonwebtoken::{decode, DecodingKey, Validation};
+use log::error;
+
+use serde::{Deserialize, Serialize};
+use std::sync::atomic::{AtomicU64, AtomicUsize, Ordering};
+
+use crate::feedback::{AppStateFeedback, TokenRecord};
+
+// Additionally, there is a short delay until a token can be used.
+// Clients need to wait that time if (for some reason) the user submitted
+// faster than limited here.
+const TOKEN_MIN_AGE: usize = 10;
```

Comment:
Do you think we need to decrease this value (to 1-2s maybe)? The idea was originally that you very likely won't do reasonable feedback within less than 10s, but now that you can edit coordinates, the feedback form opens only after selecting a coordinate.

Reply:
I don't see an immediate need:
- for the user this presents as a bit of extra wait time, but is not workflow breaking
- I timed one of my flatmates on this:  
   if you actually read the text, understanding the modal takes about this time
  (I think people read this modal, since we get delete after processing requests and people do need to check the GitHub box)

The real issue is: Why do we have a `TOKEN_MIN_AGE` in the first place? From a security perspective, I don't see a immediate upside.

Reply:
No, it doesn't directly help to do rate limiting (because clients can technically generate a new token before the previous one was used). The idea was that I couldn't see a case where a feedback token is requested (i.e. the form opened) and immediately used. Reasonable/helpful feedback that we want would take people >=10s to write and send. This was intended as some kind of spam protection (of bored humans or of bots pressing wildly on buttons; we haven't seen that yet though).

Reply:
> This was intended as some kind of spam protection (of bored humans or of bots pressing wildly on buttons; we haven't seen that yet though).

I am not convinced that bored humans are a real threat.
Bots are not an issue. Most bots are not even smart enough to execute JS...

It does not hurt us and does not add significant complexity, imo ⇒ we should keep it and lower the limit to 5s.
What do you think?

Reply:
```suggestion
const TOKEN_MIN_AGE: usize = 5;
```

Reply:
Sounds like a good idea